File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangePlayerRequestMessage.scala





















// Generated on 05/03/2013 10:34:55
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangePlayerRequestMessage extends ExchangeRequestMessage {
  override val messageId = 5773

  protected trait Message extends ExchangeRequestMessage.Message with Serializable {
    override val messageId = ExchangePlayerRequestMessage.messageId

    val target: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(target)
    }

  }

  protected def create(exchangeType: Byte, target: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val exchangeType = buf.readByte()
    val target = buf.readInt()

    create(exchangeType, target)
  }

}

object ExchangePlayerRequestMessage extends ExchangePlayerRequestMessage {
  case class MessageImpl(exchangeType: Byte, target: Int) extends Message
  type Target = MessageImpl

  override def create(exchangeType: Byte, target: Int) = MessageImpl(exchangeType, target)

  def apply(exchangeType: Byte, target: Int) = create(exchangeType, target)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangePlayerRequestMessage.java


// Generated on 05/08/2013 19:37:54
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangePlayerRequestMessage extends ExchangeRequestMessage {
    public static final int MESSAGE_ID = 5773;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int target;
    
    public ExchangePlayerRequestMessage() { }
    
    public ExchangePlayerRequestMessage(byte exchangeType, int target) {
        super(exchangeType);
        this.target = target;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(target);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        target = buf.readInt();
        if (target < 0)
            throw new RuntimeException("Forbidden value on target = " + target + ", it doesn't respect the following condition : target < 0");
    }
    
}
